node/core/approval-voting/src/lib.rs

```diff
@@ -1199,7 +1245,12 @@ async fn handle_from_overseer<Context>(
 			crate::ops::canonicalize(db, block_number, block_hash)
 				.map_err(|e| SubsystemError::with_origin("db", e))?;
 
-			wakeups.prune_finalized_wakeups(block_number);
+			// `prune_finalized_wakeups` prunes all finalized block hashes. We prune spans accordingly.
+			wakeups.prune_finalized_wakeups(block_number, &mut state.spans);
+
+			// // `prune_finalized_wakeups` prunes all finalized block hashes. We prune spans accordingly.
+			// let hash_set = wakeups.block_numbers.values().flatten().collect::<HashSet<_>>();
+			// state.spans.retain(|hash, _| hash_set.contains(hash));
```

Comment:
```suggestion
```

Reply:
`cargo fmt +nightly --all` got this one.

Reply:
Ah weird, confusing - it looks like it formatted the commented out code onto one line. I must have already commented by the time you added the suggestion 👍🏼